packages/vue-app/template/components/nuxt-link.client.js

```diff
@@ -42,29 +33,37 @@ export default {
       default: '<%= router.linkPrefetchedClass %>'
     }<% } %>
   },
+  data () {
+    return {
+      __observed: false,
+      __unobserve: () => {}
+    }
+  },
   mounted () {
     if (this.prefetch && !this.noPrefetch) {
-      this.handleId = requestIdleCallback(this.observe, { timeout: 2e3 })
+      this.handleId = requestIdleCallback(this.observe, { timeout: 2000 })
     }
   },
   beforeDestroy () {
     cancelIdleCallback(this.handleId)
 
-    if (this.__observed) {
-      observer.unobserve(this.$el)
-      delete this.$el.__prefetch
+    if (!this.__observed) {
+      return
     }
+    this.$options.__unobserve(this.$el)
+    delete this.$el.__prefetch
   },
   methods: {
     observe () {
-      // If no IntersectionObserver, avoid prefetching
-      if (!observer) {
-        return
-      }
       // Add to observer
       if (this.shouldPrefetch()) {
+        const unobserveFunctionOrUnavailable = this.$nuxt.context.observer.use(this.$el)
+        if(!unobserveFunctionOrUnavailable) {
+          return
+        }
+
+        this.$options.__unobserve = unobserveFunctionOrUnavailable
```

Comment:
If it is falsy, can we simply assign like `this.$options.__unobserve = this.$nuxt.context.observer.use(this.$el)`? (it is a `false` reference but anyway for older browsers. I think simplicity worth over it)